### ISSUE TITLE
How to annotate a high repeat contig?

### ISSUE THREAD

**Issue by YPGG1234** [Gaius-Augustus/BRAKER#141]
Hi ,
   Recently I have used  braker2 to annotate a single Y chromosome (~ 10M) , but may be it has too much region be masked (soft), therefore, I always got errors at the step of  gmes_petap.pl，and I alway get the error :

`error, no valid sequences were found`

the detail information is :

`
check before run 
create directories
commit input data
data report
commit training data
training data report
prepare initial model
get GC of sequence
GC 42
build initial ET model
running step ET_A
running gm.hmm on local multi-core system
3 contigs in training
concatenate predictions: /usr/braker/GeneMark-ET/run/ET_A_1
training level ET_A: /usr/braker/GeneMark-ET/run/ET_A_1
From 1210 loaded 1210 and ignored dublications 0
exon    no_match        match_one       match_two
Initial 619     3       0
Internal        519     7       4
Terminal        616     6       0
Single  470     0       0
CDS_no_match    all     short   long    seq_short       seq_long
CDS_no_match    2224    2101    123     763507  136881
Intergenic      all     between_match   seq_match
Intergenic:     1093    0       0
running step ET_B
running gm.hmm on local multi-core system
3 contigs in training
concatenate predictions: /usr/braker/GeneMark-ET/run/ET_B_1
training level ET_B: /usr/braker/GeneMark-ET/run/ET_B_1
From 1210 loaded 1210 and ignored dublications 0
exon    no_match        match_one       match_two
Initial 1973    3       0
Internal        2896    5       1
Terminal        1973    2       0
Single  186     0       0
CDS_no_match    all     short   long    seq_short       seq_long
CDS_no_match    7028    6902    126     1389228 137709
Intergenic      all     between_match   seq_match
Intergenic:     2162    0       0
error, no valid sequences were found
error on call: /usr/bin/gm_et_linux_64/make_nt_freq_mat.pl --cfg
`

the command line I used is : 

`braker.pl --cores 10 --species=XXX --genome=Y.fa 
     --bam=Y.rna.bam 
     --prot_seq=short_distance_protein.fa --prg=gth 
     --gff3 --min_contig=10000 --gth2traingenes`

I wonder whether can I annotate it with braker2 and what should  I do ? If not , what is a alternative way.Can you help me? Thanks.

**Comment by KatharinaHoff** [comment]
Your call is (up to failure) only affected by the bam file, not by the protein file. Please check on your braker working directory how many high coverage introns are in genemark_hintsfile.gff

`cat genemark_hintsfile.gff | perl -ne 'm/mult=(\d+)/; if($1>10){print $_;}' | wc -l `

If there are only a few hundred introns, that might cause the problem.

Another possibility is that your chromosome indeed doesn't provide sufficient amount of sequence. But it's often about the number of high coverage introns.

**Comment by KatharinaHoff** [comment]
@YPGG1234  Did you manage to work around your problem, or does it persist with the new BRAKER release?

**Comment by YPGG1234** [comment]
Dear KatharinaHoff,

I'm sorry for my late reply. Now I decided to annotate the whole genome with braker2 instead of single Y chromosome, which worked well at first, but when GenomeThreader got to the X chromosome, the status of the task does not change, and it's been going on for about 20 days now.The command line I used was as follows:

`braker.pl --cores 25 --genomes=softmasked_genome.fa
     --bam= N1.rna.bam N2.rna.bam ... N8.rna.bam
     --prot_seq=pep.fa --prg=gth 
     --gff3 --softmasking`

I know this pattern takes long time, but the autosomes run the GenomeThreader only for 4 days, so I wonder if this is due to too many repeats in X chromosome or some other reasons? Is there any solution?
